tests/warn/i22681.scala

```diff
@@ -0,0 +1,13 @@
+
+//> using options -Wunused:all
+
+trait T:
+  def t: Int
+
+class C:
+  def f: Runnable { def u: Int } = new Runnable with T:
+    private def v = 42 // avoid g judged too trivial to warn
+    def run() = ()
+    def g = v // warn effectively private member is unused
+    def t = v // nowarn
+    def u = v // warn despite structural type
```

Comment:
I assume that last warning is undesirable (false positive)? If yes, comment that it is so?

If not, why is it desirable?